examples/tutorials/introduction/1_validation.py

```diff
@@ -0,0 +1,178 @@
+"""
+Validation
+==========
+
+A common issue faced by scientists is ensuring that the data that is
+entered matches the sort of data that is expected.  For example, we expect
+that the filename is a string (or perhaps a pathlib **Path**, for advanced
+users), the operator name is a string, the acquisition date is a
+datetime.date object, and so on.  Many languages allow (or even require)
+you to specify these data types as part of your program, but even then these
+data types tend to reflect what the computer stores in memory, and not what
+the data actually _represents_.
+
+For example, not only should the file be a string, but we would like to
+validate that the string is in fact a valid path name, and ideally that the
+file can actually be found.  A sample ID may be expected to follow some pattern
+based on lab protocols.  The image data is expected to be a 2D array of values
+rather than just any NumPy array.  And so on.
+
+Traits provides a way to ensure that values are validated when assigned in your
+classes: as part of your class definition you _declare_ what you expect your
+data to be.  To do this, the first thing you want to do is inherit from the
+base ``HasTraits`` class which enables all of the validation machinery::
+
+    from traits.api import HasTraits
+
+    class Image(HasTraits):
+        ''' An SEM image stored in a file. '''
+
+Having done this, we can declare the types of data that we expect for the
+attributes of an ``Image`` class.  For example, we expect that the operator
+name should be a string, so we can use the standard Traits ``Str`` trait
+type::
+
+    from traits.api import HasTraits, Str
+
+    class Image(HasTraits):
+        ''' An SEM image stored in a file. '''
+
+        operator = Str()
+
+Now, if we try and assign any other value to the ``operator`` attribute, we
+find that the class will raise a ``TraitError``::
+
+    >>> image.operator = 3
```

Comment:
Missing a definition of `image` here. It would be worth adding the `image = Image()` line.